actions/middleware.go

```diff
@@ -101,3 +101,33 @@ func CheckAuthentication(appConfig *config.AppConfig, bux bux.ClientInterface, r
 	// Return an empty error message
 	return req, dictionary.ErrorMessage{}
 }
+
+// Request will write the request to the logs before and after calling the handler
+func Request(h httprouter.Handle, a *Action) httprouter.Handle {
+	return parameters.MakeHTTPRouterParsedReq(func(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+		// Start the custom response writer
+		guid, _ := uuid.NewV4()
+		writer := &apirouter.APIResponseWriter{
+			IPAddress:      apirouter.GetClientIPAddress(req),
+			Method:         req.Method,
+			RequestID:      guid.String(),
+			ResponseWriter: w,
+			Status:         0, // future use with E-tags
+			URL:            req.URL.String(),
+			UserAgent:      req.UserAgent(),
+		}
+
+		// Start the log (timer)
+		start := time.Now()
+
+		// Fire the request
+		h(writer, req, ps)
+
+		// Complete the timer and final log
+		elapsed := time.Since(start)
+
+		if a.AppConfig.RequestLogging {
+			a.Services.Logger.Debug().Msgf("%d | %s | %s | %s | %s ", writer.Status, elapsed, req.RemoteAddr, req.Method, req.URL)
```

Comment:
Additionally we could add those params in contextual fields and maybe some additional marker like Str("logger", "http-request")

Reply:
So I should leave it in a message and additionally add to context?

Reply:
Done, I added all parameters into logger